### PR TITLE
feat(trace): Add URL utilities for layout settings

### PR DESCRIPTION
**Pull request by Suyog241005**
## Which problem is this PR solving?
- PR 1 of 5 implementing [ADR-0010: Layout Settings Priority Stack](https://github.com/jaegertracing/jaeger-ui/blob/main/docs/adr/0010-layout-settings-priority-stack.md).

## Description of the changes
- Adds pure URL utility functions for reading and writing layout settings (`timelineBarsVisible`, `detailPanelMode`) from/to the URL query string. No React dependency; no UI changes.

**New exports in `TracePage/url/index.ts`:**
- `parseLayoutSettingsFromUrl(search)` — parses `?timeline=on|off` and `?sidebar=inline|sidepanel` into typed nullable values (`null` when absent or unrecognized)
- `stringifyLayoutSettings(settings)` — converts typed settings back into query param key/value pairs
- `stripLayoutSettings(search)` — strips all layout params from a URL while preserving unrelated params (e.g. `uiFind`)
- `stripLayoutSettingParam(search, key)` — strips only the single acted-on param; toggling the timeline switch does not silently drop an active `?sidebar=` override
- `getUrl` / `getTracePageLink` — extended with an optional `settings` argument

## How was this change tested?
- 20 unit tests covering all new functions: round-trip encoding, partial settings, `stripSettingParam` isolation (toggling timeline leaves sidebar param intact and vice versa), early-return when param is already absent, and edge cases with no params

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully: `make lint test`

## AI Usage in this PR (choose one)
See [AI Usage Policy](https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md#ai-usage-policy).
- [ ] **None**: No AI tools were used in creating this PR
- [ ] **Light**: AI provided minor assistance (formatting, simple suggestions)
- [x] **Moderate**: AI helped with code generation or debugging specific parts
- [ ] **Heavy**: AI generated most or all of the code changes

### CODE REVIEW

**Comment by Suyog241005** [comment]
Hi @yurishkuro this is Part 1/5 of the split described in [ADR-0010](https://github.com/jaegertracing/jaeger-ui/blob/main/docs/adr/0010-layout-settings-priority-stack.md).
This PR contains only the three pure URL utility functions (`parseSettingsFromUrl`, `stringifySettings`, `rebaseSettings`) with no React dependency. The rebaseSettings bug from your review is fixed here — it now strips only the specific param that was acted on rather than both.

File: packages/jaeger-ui/src/components/TracePage/url/index.ts [hunk 1]
```diff
@@ -6,30 +6,116 @@ import queryString from 'query-string';
...
+  if (settings.timelineBarsVisible === false) params[URL_PARAM_TIMELINE] = 'off';
+  if (settings.detailPanelMode) params[URL_PARAM_SIDEBAR] = settings.detailPanelMode;
+  return params;
+}
+
+/**
+ * Removes all layout setting params from a search string, preserving others (e.g. uiFind).
+ * Used when closing/navigating away from a trace to produce a clean base URL.
+ */
+export function rebaseSettings(search: string): string {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/url/index.ts]: I think the naming is too generic. The comment says "preserving others" meaning there are other URL params, but the function itself has a generic name in a generic url/index module.

Comment (Suyog241005) [reply packages/jaeger-ui/src/components/TracePage/url/index.ts]: I will rename `rebaseSettings` to `stripLayoutSettings`, is it good?

Comment (yurishkuro) [reply packages/jaeger-ui/src/components/TracePage/url/index.ts]: I didn't mean just this one function.

Comment (Suyog241005) [reply packages/jaeger-ui/src/components/TracePage/url/index.ts]: You're right, I only caught `rebaseSettings`. I've rename all four layout-settings functions in the module: `stringifySettings` → `stringifyLayoutSettings`, `stripSettingParam` → `stripLayoutSettingParam`, and `parseSettingsFromUrl` → `parseLayoutSettingsFromUrl`. Every export in url/index.ts now includes "layout" in its name, making it immediately clear from the call site what domain it operates on.

**Comment by Suyog241005** [comment]
@yurishkuro I’ve addressed all the feedback from GitHub Copilot. I’d really appreciate your review and feedback on the PR

**Comment by Suyog241005** [comment]
@yurishkuro just a gentle follow up.
